packages/arb-token-bridge-ui/src/components/MainContent/MainContent.tsx

```diff
@@ -28,161 +25,78 @@ const motionDivProps = {
-          const txCreatedAt = tx.createdAt
-
-          // Should never be the case, more of a sanity check
-          if (!txCreatedAt) {
-            dispatch({ type: 'set_tx_as_seen', payload: tx.txId })
-            return
-          }
+    // if pending deposits found, add them in the store
+    arbTokenBridge?.transactions?.setDepositsInStore?.(
+      pendingTxns?.pendingDeposits || []
+    )
 
-          // We only pick those older than the cache, so we don't accidentally mark fresh withdrawals as seen.
-          if (new Date(txCreatedAt) < cacheCreatedTimestamp) {
-            dispatch({ type: 'set_tx_as_seen', payload: tx.txId })
-          }
-        })
-    }
-    // It's safe to omit `dispatch` from the dependency array: https://reactjs.org/docs/hooks-reference.html#usereducer
-  }, [didLoadPendingWithdrawals, unseenTransactions])
+    // if pending withdrawals found, add them in the store
+    arbTokenBridge?.setWithdrawalsInStore?.(
+      pendingTxns?.pendingWithrawals || []
+    )
+  }, [pendingTxns])
 
   return (
     <div className="flex w-full justify-center">
       <div className="w-full max-w-screen-lg flex-col space-y-6">
         <AnimatePresence>
-          {unseenTransactions.map(tx =>
-            isDeposit(tx) ? (
-              <motion.div key={tx.txId} {...motionDivProps}>
-                <DepositCard key={tx.txId} tx={tx} />
-              </motion.div>
-            ) : (
-              <motion.div key={tx.txId} {...motionDivProps}>
-                <WithdrawalCard key={tx.txId} tx={tx} />
-              </motion.div>
-            )
-          )}
+          <motion.div
+            key="transfer-panel"
+            {...motionDivProps}
+            className="relative z-10"
+          >
+            <TransferPanel />
+          </motion.div>
         </AnimatePresence>
 
         <AnimatePresence>
-          {isTransferPanelVisible && (
-            <motion.div
-              key="transfer-panel"
-              {...motionDivProps}
-              className="relative z-10"
-            >
-              <TransferPanel />
+          <>
+            <motion.div key="explore-arbitrum" {...motionDivProps}>
+              <ExploreArbitrum />
             </motion.div>
-          )}
-        </AnimatePresence>
-
-        <AnimatePresence>
-          {unseenTransactions.length > 0 && (
-            <>
-              <motion.div key="explore-arbitrum" {...motionDivProps}>
-                <ExploreArbitrum />
-              </motion.div>
 
-              <div className="h-[25vh]" />
-            </>
-          )}
+            <div className="h-[25vh]" />
+          </>
         </AnimatePresence>
       </div>
+
+      <SidePanel
+        isOpen={isTransactionHistoryPanelVisible}
+        heading="Transaction History"
+        onClose={closeTransactionHistory}
+      >
+        <div className="flex flex-col justify-around gap-6">
+          {/* Pending transactions cards */}
+          <PendingTransactions
```

Comment:
Yeah, that was being done initially, but then I realized we need to fetch the pending transactions only once on App load and not on this component mount. Otherwise, it would keep on re-fetching pending transactions when the panel was opening. 
Hence the state of this component is kept outside of it.